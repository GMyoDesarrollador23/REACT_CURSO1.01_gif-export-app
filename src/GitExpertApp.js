import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GitExpertApp = ({
   defaultCategories = [],
}) => {
   const [categories, setCategories] = useState(
      defaultCategories
   );

   return (
      <>
         <h2>GitExpertApp</h2>
         <AddCategory setCategories={setCategories} />
         <hr />

         <ol>
            {categories.map((category) => (
               <GifGrid
                  key={category}
                  category={category}
               />
            ))}
         </ol>
      </>
   );
};
