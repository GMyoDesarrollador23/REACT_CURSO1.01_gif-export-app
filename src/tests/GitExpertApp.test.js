import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";

import { GitExpertApp } from "../GitExpertApp";
describe("Pruebas sobre el <GitExpertApp/>", () => {
   // ------------------------------------------------------
   test("debe mostrarse correctante ", () => {
      const wrapper = shallow(<GitExpertApp />);
      expect(wrapper).toMatchSnapshot();
   });

   // ------------------------------------------------------
   test("debe mostrar una lista de categorias", () => {
      const categories = ["one punch", "dragon ball"];
      const wrapper = shallow(
         <GitExpertApp defaultCategories={categories} />
      );
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find("GifGrid").length).toBe(
         categories.length
      );
   });
});
