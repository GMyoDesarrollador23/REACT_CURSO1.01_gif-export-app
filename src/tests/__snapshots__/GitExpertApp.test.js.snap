// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el <GitExpertApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre el <GitExpertApp/> debe mostrarse correctante  1`] = `
<Fragment>
  <h2>
    GitExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
